### PR TITLE
Adding sorting networks lessons (fixes #458)

### LINKED ISSUE #458: Add sorting networks lessons
- [ ] Lesson 2 (8-10) - needs generated resources
- [ ] Lesson 1 (11-14)
- [ ] Lesson 2 (11-14)

### PR DESCRIPTION
**Pull request by hayleyavw**
~~Have not added the generated resources yet, the lesson content is ready to review though.~~

Sorting Network Cards resource has now been updated, @tah942 can you please check the new card types (words, letters, Māori colours) are as you expected when reviewing this PR.

Note @JackMorganNZ this has unveiled a bug where some exercises are being listed multiple times, probably related to Sorting Networks having lessons used in multiple age groups.

screen shot 2017-06-18 at 11 45 23 am

### CODE REVIEW

File: csunplugged/templates/resources/sorting-network-cards.html [hunk 1]
```diff
@@ -35,5 +35,23 @@
         Māori numbers
       </label>
     </div>
+    <div class="form-check">
+      <label class="form-check-label">
+        <input type="radio" class="form-check-input" name="type" value="words" checked="checked">
```
Comment (JackMorganNZ) [inline csunplugged/templates/resources/sorting-network-cards.html]: This has a duplicate `checked` value, which should be removed.

File: csunplugged/topics/content/en/sorting-networks/unit-plan/lessons/investigating-variations-using-the-sorting-network.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+{image file-path="img/topics/sorting-network-crochet-v-crocodile.png"}
+
+Comparing two words or names is challenging; they will need to know to compare each character until two differ (e.g. for "crochet" and "crocodile", the "croc" prefix is the same, so it is the "h" and "o" that determine their order; this process is an algorithm in itself!)
+
+{image file-path="img/topics/sorting-network-variation-words-2.png"}
+
+The words being compared could also be used to reinforce spelling or meaning; for example, the words above are the colours in Te Reo Māori, so the student with the word "kowhai" would be reinforcing that it means the colour yellow.
+The use of macrons and other diacritical marks also gives the opportunity to explore the order that is used in the such languages for those letters.
+
+### Variation 5: Music notation
```
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/sorting-networks/unit-plan/lessons/investigating-variations-using-the-sorting-network.md]: There is two "Variations 5"s.

**Comment by hayleyavw** [comment]
@JackMorganNZ this is ready for re-review.

I changed the name of the image used on the homepage to make it more obvious that it shouldn't be moved, hope that's ok?

**Comment by hayleyavw** [comment]
@JackMorganNZ forgot to move the ct links to a separate file! this will need reviewing again.

File: csunplugged/resources/views/sorting_network_cards.py [hunk 3]
```diff
@@ -47,25 +47,24 @@ def resource_image(request, resource):
...
+            "tahi", "rua", "toru", "whā", "rima", "ono", "whitu", "waru",
+            "iwa", "tekau", "tekau mā tahi", "tekau mā waru", "tekau mā toru",
+            "tekau mā whā", "rua tekau", "rua tekau mā ono"
+        ]
+    elif card_type == "words":
+        font_size = 500
+        text = ["crocodile", "crochet", "kiwi", "weka", "kiwi", "kiwano"]
+    elif card_type == "letters":
+        font_size = 800
+        text = ["Z", "O", "N", "K", "E", "D", "S", "P", "G", "B", "I", "Y"]
```
Comment (tah942) [inline csunplugged/resources/views/sorting_network_cards.py]: on line 62 replace the Z with an L

File: csunplugged/resources/views/sorting_network_cards.py [hunk 4]
```diff
@@ -47,25 +47,24 @@ def resource_image(request, resource):
...
+    else:
+        font_size = 500
+        text = [
+            "whero", "kākāriki", "kiwikiwi", "karaka",
+            "kowhai", "pango", "māwhero", "mā"
```
Comment (tah942) [inline csunplugged/resources/views/sorting_network_cards.py]: On line 67 update kowhai to have a macron above the o.

**Pull request merged**